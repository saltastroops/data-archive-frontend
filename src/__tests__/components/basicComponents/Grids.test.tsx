import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import {
  ButtonGrid,
  DataGrid,
  MainGrid,
  ParentGrid,
  ProposalGrid,
  SubGrid,
  TargetGrid,
  TelescopeGrid
} from "../../../components/basicComponents/Grids";

describe("Grids ", () => {
  it("should render all grids", () => {
    expect(shallow(<ParentGrid />)).toBeDefined();
    expect(shallow(<TargetGrid />)).toBeDefined();
    expect(shallow(<ProposalGrid />)).toBeDefined();
    expect(shallow(<DataGrid />)).toBeDefined();
    expect(shallow(<TelescopeGrid />)).toBeDefined();
    expect(shallow(<ButtonGrid />)).toBeDefined();
    expect(shallow(<MainGrid />)).toBeDefined();
    expect(shallow(<SubGrid />)).toBeDefined();
  });

  it("should render all grids correctly", () => {
    // Use mount instead of shallow for better snapshots
    expect(toJson(mount(<ParentGrid />))).toMatchSnapshot();
    expect(toJson(mount(<TargetGrid />))).toMatchSnapshot();
    expect(toJson(mount(<ProposalGrid />))).toMatchSnapshot();
    expect(toJson(mount(<DataGrid />))).toMatchSnapshot();
    expect(toJson(mount(<TelescopeGrid />))).toMatchSnapshot();
    expect(toJson(mount(<ButtonGrid />))).toMatchSnapshot();
    expect(toJson(mount(<MainGrid />))).toMatchSnapshot();
    expect(toJson(mount(<SubGrid />))).toMatchSnapshot();
  });
});
