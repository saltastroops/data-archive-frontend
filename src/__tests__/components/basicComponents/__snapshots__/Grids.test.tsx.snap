// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grids  should render all grids correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: auto auto;
  padding: 10px;
  gap: 20px;
}

@media (max-width:900px) {
  .c0 {
    grid-template-columns: auto;
  }
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "grid-container",
            "disable": true,
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  display: grid;
  grid-template-columns: auto auto;
  padding: 10px;
  gap: 20px;

  @media (max-width: 900px) {
    grid-template-columns: auto;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="grid-container c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 2`] = `
.c0 {
  background-color: #f6f7e1;
  border: 1px solid rgba(246,247,225,0.5);
  text-align: center;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "grid-item target-form",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: #f6f7e1;
  border: 1px solid rgba(246, 247, 225, 0.5);
  text-align: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="grid-item target-form c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 3`] = `
.c0 {
  background-color: #f2fffb;
  border: 1px solid rgba(242,255,251,0.5);
  text-align: center;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "grid-item proposal-form",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: #f2fffb;
  border: 1px solid rgba(242, 255, 251, 0.5);
  text-align: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="grid-item proposal-form c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 4`] = `
.c0 {
  background-color: #e0edf9;
  border: 1px solid rgba(224,237,249,0.5);
  text-align: center;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "grid-item data-form",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: #e0edf9;
  border: 1px solid rgba(224, 237, 249, 0.5);
  text-align: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="grid-item data-form c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 5`] = `
.c0 {
  background-color: #fff4fa;
  border: 1px solid rgba(255,244,250,0.5);
  text-align: center;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "grid-item telescope-form",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-color: #fff4fa;
  border: 1px solid rgba(255, 244, 250, 0.5);
  text-align: center;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="grid-item telescope-form c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 6`] = `
.c0 {
  border: 1px solid rgba(255,255,255,0);
  text-align: left;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "grid-item",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  border: 1px solid rgba(255, 255, 255, 0);
  text-align: left;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="grid-item c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 7`] = `
.c0.c0 {
  display: grid;
  grid-template-columns: 50% 50%;
  padding: 10px;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "main-grid",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  && {
    display: grid;
    grid-template-columns: 50% 50%;
    padding: 10px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="main-grid c0"
    />
  </StyledComponent>
</styled.div>
`;

exports[`Grids  should render all grids correctly 8`] = `
.c0.c0 {
  text-align: left;
  padding-left: 10px;
}

<styled.div>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "sub-grid",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-dnqmqq",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  && {
    text-align: left;
    padding-left: 10px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-dnqmqq",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="sub-grid c0"
    />
  </StyledComponent>
</styled.div>
`;
